core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java

```diff
@@ -63,9 +64,16 @@ public String getName() {
 
         @Override
         public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient client) throws IOException {
+            validateOpType(request.params().get("op_type"));
             request.params().put("op_type", "create");
             return RestIndexAction.this.prepareRequest(request, client);
         }
+
+        public void validateOpType(String opType) {
```

Comment:
maybe make this static and pkg private?

Reply:
@s1monw what about private (or pkg private) and final?
It cannot be set to `static` as the inner class `CreateHandler` itself is not static.